security/src/test/groovy/org/aggregatortech/dindora/security/authentication/providers/FileIDProviderTest.groovy

```diff
@@ -0,0 +1,192 @@
+        File f = new File("./idstore");
+        f.createNewFile()
+
+        BufferedWriter writer = new BufferedWriter(new FileWriter(f));
+        String jsonStr = "atul-welcome1";
+        writer.append(jsonStr);
+        writer.close();
+
+        helper = ServiceLocatorHelper.getServiceLocator().getService(SystemHelper.class);
+        helper.writeProperty(FileIDProvider.IDSTORE_LOC,"./idstore")
+
+        then :
+
+            try {
+                idp.configure();
+                assert false
+            }
+        catch (IDStoreNotConfiguredException ex) {
+            assert ex.getCause() instanceof JsonParseException
+
+        }
+
+        when :
+        writer = new BufferedWriter(new FileWriter(f));
+        jsonStr = "{ \"users\": \"atul\"}";
+        writer.append(jsonStr);
+        writer.close();
+
+        then :
+        try {
+            idp.configure();
+        }
+        catch (IDStoreNotConfiguredException idEx) {
+            assert  idEx.getErrorCode().contains(MessageService.IDSTORE_NOT_CONFIGURED)
+
+
+        }
+
+        when :
+        writer = new BufferedWriter(new FileWriter(f));
+        jsonStr = "{ \"users\": [{ \"userid\":\"atul\" , \"password\":\"welcome1\" }, " +
+          "{ \"userid\":\"ajeet\" , \"password\":\"welcome1\" } ]}";
+        writer.append(jsonStr);
+        writer.close();
+
+        then :
+        try {
+            idp.configure();
+        }
+        catch (IDStoreNotConfiguredException idEx) {
+            assert  idEx.getCause() instanceof  UnrecognizedPropertyException
+
+
+        }
+        cleanup:
+        f.delete();
+
+    }
+
+    def "Test Configure  is success if format of json is an array of auth creds "() {
+
+        setup:
+
+        when :
+        IDProvider idp = new FileIDProvider();
+        File f = new File("./idstore");
+        f.createNewFile()
+        BufferedWriter writer = new BufferedWriter(new FileWriter(f));
+        String jsonStr = "{ \"users\": [{ \"username\":\"atul\" , \"password\":\"welcome1\" }, " +
+                "{ \"username\":\"ajeet\" , \"password\":\"welcome1\" } ]}";
+        writer.append(jsonStr);
+        writer.close();
+
+        then :
+
+
+            idp.configure();
+
+
+        cleanup:
+        f.delete();
+    }
+    def "Test Authenticate method"() {
+
+        setup:
+
+        when :
+        IDProvider idp = new FileIDProvider();
+        File f = new File("./idstore");
+        f.createNewFile()
+        BufferedWriter writer = new BufferedWriter(new FileWriter(f));
+        String jsonStr = "{ \"users\": [{ \"username\":\"atul\" , \"password\":\"welcome1\" }, " +
+                "{ \"username\":\"ajeet\" , \"password\":\"welcome\" } ]}";
+        writer.append(jsonStr);
+        writer.close();
+        idp.configure();
+        then :
+
+        AuthenticationCredentials authCreds = new UserNamePasswordCredentials("atul1",'welcome1');
+       assert  idp.authenticate(authCreds) == false
```

Comment:
Shouldn't this be true in this case?

Reply:
its atul1 while the actual user is atul

Reply:
My bad, I was looking for a positive test case and my mind saw what it wanted to see:)